source/material_model/grain_size.cc

```diff
@@ -1069,7 +1069,19 @@ namespace aspect
           if (prm.get ("Material file format") == "perplex")
             material_file_format = perplex;
           else if (prm.get ("Material file format") == "hefesto")
-            material_file_format = hefesto;
+            {
+              material_file_format = hefesto;
+              AssertThrow (material_file_names.size() == derivatives_file_names.size(),
```

Comment:
Thanks for improving the error message, but this condition only needs to be met if the input parameter `Use enthalpy for material properties` is set to true (otherwise the derivative files are not used). The parameter is already available in this place of the code as the variable `use_enthalpy`, so the following should be correct:
```suggestion
              AssertThrow (use_enthalpy == false || material_file_names.size() == derivatives_file_names.size(),
```